### PR TITLE
Update release.md

### PR DESCRIPTION
**Pull request by wenjiaswe**
Update release.md based on 3.5.15 release: https://github.com/etcd-io/etcd/issues/18247#issuecomment-2240080965

### CODE REVIEW

**Comment by wenjiaswe** [comment]
cc @jmhbnz @ivanvc

File: Documentation/contributor-guide/release.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ which don't need to be executed before releasing each version.
 4. Authenticate the image registry, refer to [Authentication methods](https://cloud.google.com/container-registry/docs/advanced-authentication).
    - `gcloud auth login`
    - `gcloud auth configure-docker`
+5. Make sure your Linux machine has at least 5GB free space.
```
Comment (ivanvc) [inline Documentation/contributor-guide/release.md:65]: Hi @wenjiaswe, I suggest rather changing pre-req 2c, that says:

> Ensure there is at least 500MB of free space on your Linux machine.

File: Documentation/contributor-guide/release.md [hunk 2]
```diff
@@ -79,13 +80,14 @@ which don't need to be executed before releasing each version.
...
    - Set the release title as the version name
+   - Choose the correct release tag (generated from step #4)
    - Follow the format of previous release pages
    - Attach the generated binaries and signature file
    - Select whether it's a pre-release
    - Publish the release
 6. Announce to the etcd-dev googlegroup
 
-   Follow the format of previous release emails sent to etcd-dev@googlegroups.com, see an example below,
+   Follow the format of previous release emails sent to etcd-dev@googlegroups.com, see an example below. After sending out the email, ask one of the mainterners to approve the email from the pending list.
```
Comment (jmhbnz) [inline Documentation/contributor-guide/release.md:90]: Typo but lets also tweak it to "mailing list maintainers" as this doesn't necessarily have to only be project code maintainers in future 🙂 

```suggestion
   Follow the format of previous release emails sent to etcd-dev@googlegroups.com, see an example below. After sending out the email, ask one of the mailing list maintainers to approve the email from the pending list.
```

**Review by serathius — APPROVED** [review]

**Pull request closed without merging**